### PR TITLE
[CVP-1214] imporve logging of deploy operator olm playbooks

### PR DESCRIPTION
**Pull request by samvarankashyap**
The following improvement logs the error tasks to json file
such that it can be consumed by the pipeline. Further,
displays the failed task and its output.

### CODE REVIEW

**Comment by samvarankashyap** [comment]
@sonam1412 I wonder why tests arent getting triggered.

**Comment by sonam1412** [comment]
> @sonam1412 I wonder why tests arent getting triggered.

can see PR#47 has test results on it, will try to re-trigger

File: roles/deploy_olm_operator/tasks/main.yml [hunk 1]
```diff
@@ -1,206 +1,241 @@
----
-- set_fact:
+- name: "Set quay relase quay release"
```
Comment (dirgim) [inline roles/deploy_olm_operator/tasks/main.yml:1]: Nit: typo 
```suggestion
- name: "Set quay release"
```

File: roles/deploy_olm_operator/tasks/main.yml [hunk 2]
```diff
@@ -1,206 +1,241 @@
...
+    - name: "Set failure result"
+      set_fact:
+        deploy_olm_operator_results: {'result': 'fail', 'name': "{{ ansible_failed_task.name }}", 'msg': "{{ ansible_failed_result }}" }
+
+    - name: "Failed on deploy_olm_operator_results"
+      fail:
+        msg: "{{ deploy_olm_operator_results }}"
+  
+  always:
+    - name: "Write parse_operator_metadata results to json file"
```
Comment (dirgim) [inline roles/deploy_olm_operator/tasks/main.yml:238]: Nit: change task name to reflect the role
```suggestion
    - name: "Write deploy_olm_operator_results results to json file"
```

File: roles/deploy_olm_operator/tasks/main.yml [hunk 3]
```diff
@@ -1,206 +1,241 @@
...
+
+    - name: "Output the InstallPlan debug output to a debug file"
+      copy:
+        content: "{{ installplan_result.stdout }}"
+        dest: "{{ work_dir }}/olm-installplan-debug.txt"
+      when: installplan_result.stdout is defined
+    
+    - name: "Set fact for deploy olm operator results"
+      set_fact:
+        deploy_olm_operator_results: {}
```
Comment (dirgim) [inline roles/deploy_olm_operator/tasks/main.yml:208]: Don't we want to fill out the results json like in the [parsing role](https://github.com/redhat-operator-ecosystem/operator-test-playbooks/blob/master/roles/parse_operator_metadata/tasks/main.yml#L174)? e.g.
```suggestion
        deploy_olm_operator_results: {'result': 'pass', 'msg': ''}
```

**Review by dirgim — CHANGES_REQUESTED** [review]

**Comment by samvarankashyap** [comment]
@dirgim PTAL

**Comment by samvarankashyap** [comment]
@sonam1412 @dirgim PTAL

**Review by dirgim — APPROVED** [review]
LGTM

**Pull request merged**